src/routes/api/api-router.ts

```diff
@@ -43,78 +28,6 @@ ApiRouter.use(rateLimit({
 	max: 60 // limit each IP to 60 requests per windowMs
 }));
 
-// All routes require a PAT to belong to someone on staff
-// This middleware will take the token and make a request to GraphQL
-// to see if it belongs to someone on staff
-
-ApiRouter.use(
-	async (req: Request, res: Response, next: NextFunction): Promise<void> => {
-		const token = req.get("Authorization");
-		if (!token) {
-			res.sendStatus(404);
-			return;
-		}
-		try {
-			// Create a separate octokit instance than the one used by the app
-			const octokit = GithubAPI({
-				auth: token.split(" ")[1]
-			});
-			const { data, errors } = (
-				await octokit.request({
-					headers: {
-						Accept: "application/json",
-						"Content-Type": "application/json"
-					},
-					method: "POST",
-					// 'viewer' will be the person that owns the token
-					query: viewerPermissionQuery,
-					url: "/graphql"
-				})
-			).data;
-
-			req.addLogFields({ login: data?.viewer?.login });
-
-			if (errors) {
-				res.status(401).json({ errors, viewerPermissionQuery });
-				return;
-			}
-
-			if (!validAdminPermission(data.viewer)) {
-				req.log.info(
-					{
-						login: data.viewer.login,
-						isAdmin: data.viewer.organization?.viewerCanAdminister
-					},
-					`User attempted to access admin API routes.`
-				);
-				res.status(401).json({
-					error: "Unauthorized",
-					message: "Token provided does not have required access"
-				});
-				return;
-			}
-
-			req.log.info({
-				req,
-				login: data.viewer.login,
-				isAdmin: data.viewer.organization?.viewerCanAdminister
-			},
-			"Admin API routes accessed"
-			);
-
-			next();
-		} catch (err) {
-			req.log.info({ err });
-
-			if (err.status === 401) {
-				res.status(401).send(err.HttpError);
-				return;
-			}
-			res.sendStatus(500);
-		}
-	}
-);
```

Comment:
By removing this, you're making the API completely open.  I think the `/api` route is being blocked by a higher level routing at the micros/infrastructure level, but it still irks me that we don't do a single check at this level (like a header check or something) because if the infrastructure changes by mistake, how will we know that our admin api is open?

Reply:
also, what happens if we change the API route in the future to `/admin` or something?

Reply:
>> also, what happens if we change the API route in the future to /admin or something?
If we do it we won't be able to call it unless we change POCO config accordingly. All open endpoints need to be explicitly whitelisted


We can address adding extra protection layer later

Reply:
SLAuth adds headers to the incoming request, which tell what authentication mechanism've been used.

We can add the middleware which verifies that these headers present there. However, these headers don't contain any token or signature we can verify to ensure that these headers were added by Service Proxy. 

However adding a check to make sure that these headers present and Authentication Mechanism was used will be able to protect us from accidentially making them open in POCO config. 

I'll create a ticket

Reply:
Done, ticket created. Will do it shortly after we roll out the changes

Reply:
[ARC-1282] for future reference